### PR TITLE
DACT-2192: error back links fix

### PR DESCRIPTION
**Pull request by anona-w**
Fix for back link navigation issues when errors are shown on name, nationality, and occupation paages

### CODE REVIEW

**Review by cdrainCH — DISMISSED** [review]

File: src/validation/name.validation.ts [hunk 1]
```diff
@@ -33,11 +33,14 @@ export const nameValidator = async (req: Request, res: Response, next: NextFunct
         const isUpdate = req.path.includes("update-director-name");
         const frontendValidationErrors = validateName(req, NameValidation, isUpdate);
         let currentUrl;
+        let backLinkUrl: string;
         if(isUpdate){
```
Comment (AndreKramer7654) [inline src/validation/name.validation.ts:37]: ident

File: src/validation/nationality.validation.ts [hunk 2]
```diff
@@ -27,20 +27,23 @@ export const nationalityValidator = async (req: Request, res: Response, next: Ne
     const nationality1 = getField(req, DirectorField.NATIONALITY_1);
     const nationality2 = getField(req, DirectorField.NATIONALITY_2);
     const nationality3 = getField(req, DirectorField.NATIONALITY_3);
-    const isUpdate = req.path.includes("update-director-name");
+    const isUpdate = req.path.includes("update-director-name") || req.path.includes("update-director-nationality");
```
Comment (AndreKramer7654) [inline src/validation/nationality.validation.ts:30]: Should this be just checking for nationality? and was wrongly checking for name so should check no longer?

File: src/validation/name.validation.ts (hunk 1 above)
Comment (anona-w) [reply src/validation/name.validation.ts:37]: indented

File: src/validation/nationality.validation.ts (hunk 2 above)
Comment (anona-w) [reply src/validation/nationality.validation.ts:30]: yep, think this was an existing error - have removed the name section

[the remark this answers is not in the document]
Comment (anona-w) [reply src/validation/nationality.validation.ts:30]: As above - think this was an existing error - have removed the name section

File: test/controllers/update/update.director.name.controller.unit.ts [hunk 3]
```diff
@@ -298,6 +305,14 @@ describe("Update Director name controller tests", () => {
         expect(response.text).not.toContain("Found. Redirecting to " + DIRECTOR_NAME_URL);
       });
 
+      it("should set back link correctly if there are errors", async () => {
```
[the remark this answers is not in the document]
Comment (anona-w) [reply test/controllers/update/update.director.name.controller.unit.ts:308]: Updated to send an invalid char as first name in the test to ensure an error is generated - nice catch

**Review by AndreKramer7654 — APPROVED** [review]

**Review by PhilipGlover — APPROVED** [review]
👍

**Review by jditchburn — APPROVED** [review]

**Pull request merged**